File: SphrLibrary/Entities/OpenmHealth/BloodPressureBody.cs
using System.Runtime.Serialization;

namespace SphrLibrary.Entities.OpenmHealth
{
    /// <summary>
    /// Open mHealth 血圧データボディ部のリソースを表します。
    /// </summary>
    [DataContract()]
    [Serializable()]
    public class BloodPressureBody : OmhBodyEntityBase
    {
        [DataMember()]
        public ValueUnit systolic_blood_pressure = new ValueUnit();
        [DataMember()]
        public ValueUnit diastolic_blood_pressure = new ValueUnit();

        [DataMember(EmitDefaultValue = false)]
        public string? body_posture = null; //BodyPostureTypeEnum
        [DataMember(EmitDefaultValue = false)]
        public string? measurement_location = null; //BloodPressureMeasurementLocationTypeEnum
        [DataMember(EmitDefaultValue = false)]
        public string? temporal_relationship_to_physical_activity = null; //TemporalRelationshipToPhysicalActivityTypeEnum
        [DataMember(EmitDefaultValue = false)]
        public string? descriptive_statistic = null;

        public BloodPressureBody() : base() { }
    }
}

File: References/Java/SphrLibrary-Java/lib/src/main/java/sphrlibrary/entities/openmhealth/BloodPressureBody.java
package sphrlibrary.entities.openmhealth;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.ToString;

/** 血圧データの body */
@Data
@NoArgsConstructor
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class BloodPressureBody extends OmhBodyEntityBase {

    /** 最高血圧 */
    @NonNull
    private ValueUnit systolicBloodPressure = new ValueUnit();

    /** 最低血圧 */
    @NonNull
    private ValueUnit diastolicBloodPressure = new ValueUnit();

    /** 測定時の身体の姿勢 */
    @NonNull
    private String bodyPosture = "";

    /** 測定位置 */
    @NonNull
    private String measurementLocation = "";

    /** 身体活動との時間的関係 */
    @NonNull
    private String temporalRelationshipToPhysicalActivity = "";

    /** 統計情報 */
    @NonNull
    private String descriptiveStatistic = "";
}
